File: datamodel/src/main/java/datamodel/clinical/treatment/history/StopReason.java
package datamodel.clinical.treatment.history;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public enum StopReason {
    PROGRESSIVE_DISEASE,
    TOXICITY;

    @Nullable
    public static StopReason createFromString(@NotNull String input) {
        String uppercase = input.toUpperCase();
        if (uppercase.contains("PD") || uppercase.contains("PROGRESSIVE")) {
            return PROGRESSIVE_DISEASE;
        } else if (uppercase.contains("TOXICITY")) {
            return TOXICITY;
        } else {
            return null;
        }
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/treatment/history/StopReason.kt
package com.hartwig.actin.datamodel.clinical.treatment.history

enum class StopReason {
    PROGRESSIVE_DISEASE,
    TOXICITY;

    companion object {
        fun createFromString(input: String): StopReason? {
            val uppercase = input.uppercase()
            return if (uppercase.contains("PD") || uppercase.contains("PROGRESSIVE")) {
                PROGRESSIVE_DISEASE
            } else if (uppercase.contains("TOXICITY")) {
                TOXICITY
            } else {
                null
            }
        }
    }
}
